services/worker/src/worker/job_runners/config/parquet_and_info.py

```diff
@@ -0,0 +1,965 @@
+        )
+
+    config_names = {config_name_item["config"] for config_name_item in config_names_content["config_names"]}
+    if config not in config_names:
+        raise ConfigNamesError(f"{config=} does not exist in {dataset=}")
+
+    hf_api = HfApi(endpoint=hf_endpoint, token=hf_token)
+    committer_hf_api = HfApi(endpoint=hf_endpoint, token=committer_hf_token)
+
+    # prepare the parquet files locally
+    local_parquet_files: List[ParquetFile] = []
+    download_config = DownloadConfig(delete_extracted=True)
+    try:
+        builder = load_dataset_builder(
+            path=dataset,
+            name=config,
+            revision=source_revision,
+            use_auth_token=hf_token,
+            download_config=download_config,
+        )
+    except _EmptyDatasetError as err:
+        raise EmptyDatasetError(f"{dataset=} is empty.", cause=err) from err
+    raise_if_too_big_from_external_data_files(
+        builder=builder,
+        max_dataset_size=max_dataset_size,
+        max_external_data_files=max_external_data_files,
+        hf_token=hf_token,
+    )
+    builder.download_and_prepare(file_format="parquet")  # the parquet files are stored in the cache dir
+    dataset_info = asdict(builder.info)
+    local_parquet_files.extend(
+        ParquetFile(local_file=local_file, local_dir=builder.cache_dir, config=config)
+        for local_file in glob.glob(f"{builder.cache_dir}**/*.parquet")
+    )
+
+    # create the target revision if it does not exist yet (clone from initial commit to avoid cloning all repo's files)
+    try:
+        refs = hf_api.list_repo_refs(repo_id=dataset, repo_type=DATASET_TYPE)
+        if all(ref.ref != target_revision for ref in refs.converts):
+            initial_commit = hf_api.list_repo_commits(repo_id=dataset, repo_type=DATASET_TYPE)[-1].commit_id
+            committer_hf_api.create_branch(
+                repo_id=dataset, branch=target_revision, repo_type=DATASET_TYPE, revision=initial_commit
+            )
+    except RepositoryNotFoundError as err:
+        raise DatasetNotFoundError("The dataset does not exist on the Hub.") from err
+
+    target_dataset_info = hf_api.dataset_info(repo_id=dataset, revision=target_revision, files_metadata=False)
+    # - get repo parquet files
+    all_repo_files: Set[str] = {f.rfilename for f in target_dataset_info.siblings}
+    repo_parquet_files: Set[str] = {file for file in all_repo_files if file.endswith(".parquet")}
+    # - get parquet files for current config
+    config_files_to_add: Dict[str, str] = {
+        parquet_file.repo_file(): parquet_file.local_file for parquet_file in local_parquet_files
+    }
+    # - get files that will be preserved in repo: files belonging to other configs and .gitattributes
+    files_to_ignore: Set[str] = {
+        file for config in config_names for file in repo_parquet_files if file.startswith(f"{config}/")
```

Comment:
I think that we should exclude the current config, because if a previous version of parquet files exists in the current config directory, but with a different set of splits, we could have dangling old files among the new files we will upload.

Reply:
so, something like:

```suggestion
        file for other_config in config_names for file in repo_parquet_files if other_config != config and file.startswith(f"{config}/")
```

(not tested, and the style will surely make the CI fail)